test/runtests.jl

```diff
@@ -123,9 +190,35 @@ end
                     -0.06772589439264813 6.670664781664879e-5])
     @test isapprox(first(predict(lm_model)), 357.57694841780994)
     @test isapprox(loglikelihood(lm_model), -4353.946729075838)
+    @test isapprox(loglikelihood(lm_model.rr), -4353.946729075838)
     @test isapprox(loglikelihood(glm_model), -4353.946729075838)
     @test isapprox(nullloglikelihood(lm_model), -4984.892139711452)
     @test isapprox(mean(residuals(lm_model)), -5.412966629787718)
+    @test GLM.working_residuals(lm_model) ≈ residuals(lm_model)
+    @test r2(lm_model) ≈ 0.8330258148644486
+    @test adjr2(lm_model) ≈ 0.832788298242634
+
+    lm_model = fit(LinearModel, f, df; wts=uweights(0))
+    @test_logs (:warn,
+                "Using `wts` of zero length for unweighted regression is deprecated in favor of " *
+                "explicitly using `UnitWeights(length(y))`." *
+                " Proceeding by coercing `wts` to UnitWeights of size $(N).")
```

Comment:
@gragusa I had missed this, but it turns out it's a no-op. And if I fix it to check the logs from the line above, it fails because we don't print a warning for `uweights(0)`. Do you think we can just remove it?

Below the situation is similar but not exactly the same. The tests for `loglikelihood(lm1)` are duplicated, and we can't check the deprecation and test pweights at the same time.